mlos_bench/README.md

```diff
@@ -47,152 +54,118 @@ With a [JSON5](https://spec.json5.org) [config file](./mlos_bench/config/) and c
+- Storage for experimental data ([`storage/sqlite.jsonc`](./mlos_bench/config/storage/sqlite.jsonc));
 
-        "tunable_values": [                              // Key/value pairs of tunable parameters. Uses config_path
-            "tunable-values-example.json"
-        ],
+They also refer to other configs, e.g.
 
-        "globals": [                                     // Config generated at step 2. Uses config_path
-            "global_config.json"
-        ],
+- Reusable config snippets in `"config_path"` section, and
+- Additional config files containing sensitive data like DB passwords and Azure credentials.
 
-        "teardown": false,                               // Do not shutdown/deprovision a VM
+> Make sure that the files `./global_config_azure.json` and `./global_config_storage.json` you created in steps 3 and 4 are included in the `"globals"` section of your CLI config.
 
-        "log_file": "os-autotune.log",                   // Log file (also prints to stdout)
-        "log_level": 10,                                 // Log level = DEBUG
+For the purpose of this tutorial, we will assume that we reuse the existing [`azure-redis-bench.jsonc`](./mlos_bench/config/cli/azure-redis-bench.jsonc) and [`azure-redis-opt.jsonc`](./mlos_bench/config/cli/azure-redis-opt.jsonc) configurations without any changes.
+In a more realistic scenario, however, you might need to change and/or create new config files for some parts of your benchmarking environment.
+We'll give more details on that below.
 
-        "experimentId": "RedisBench",                    // Experiment ID (can be in global_config.json)
-        "trialId": 1                                     // Trial ID (can come from the persistent storage service)
-    }
-    ```
+### 5. Create another config file for the parameters specific to your experiment
 
-8. Run our configuration through `mlos_bench`:
+Copy one of our examples, e.g., [`experiment_RedisBench.jsonc`](./mlos_bench/config/experiments/experiment_RedisBench.jsonc) and name it after your experiment, e.g. `experiment_MyBenchmark.jsonc`.
 
-    ```sh
-    mlos_bench --config "config_bench.jsonc"
-    ```
+In that file, you can specify any parameters that occur in your other configs, namely in `"const_args"` section of the Environment configs, or in `"config"` sections of your Service, Storage, or Optimizer configurations.
 
-    This should run a single trial with the given tunable values, write the results to the log and keep the environment running (as directed by the `"teardown": false` configuration parameter).
+> A general rule is that the parameters from the global configs like `./global_config_azure.json` or `experiment_MyAppBench.jsonc` override the corresponding parameters in other configurations.
+> That allows us to propagate the values of the parameters that are specific to the experiment into other components of the framework and keep the majority of the config files in our library immutable and reusable.
```

Comment:
This is a really important detail to document, thank you!

We might also want to include this in the `configs/README.md` section: basically just some general overview of how content flows from one config to another so that people can have some principles by which to construct their own configs reasonably well without reading the code.

Reply:
Added this.